{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SHARE Query Requests from the Community\n",
    "\n",
    "Here's where we can keep track of code for common things that members of the SHARE community might like to know!\n",
    "\n",
    "## Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import furl\n",
    "import json\n",
    "import requests\n",
    "\n",
    "SHARE_SEARCH_API = 'https://osf.io/api/v1/share/search/'\n",
    "ALL_PROVIDER_INFO = requests.get('https://osf.io/api/v1/share/providers/').json()['providerMap']\n",
    "\n",
    "def query_share(url, query):\n",
    "    headers = {'Content-Type': 'application/json'}\n",
    "    data = json.dumps(query)\n",
    "    return requests.post(url, headers=headers, data=data, verify=False).json()\n",
    "\n",
    "def get_longname_for_shortname(shortname):\n",
    "    for source in ALL_PROVIDER_INFO.keys():\n",
    "        if source == shortname:\n",
    "            return ALL_PROVIDER_INFO[source]['long_name']\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/erin/.virtualenvs/tuts/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html\n",
      "  InsecureRequestWarning)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_shortname</th>\n",
       "      <th>source_longname</th>\n",
       "      <th>earliest_date</th>\n",
       "      <th>latest_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>datacite</td>\n",
       "      <td>DataCite MDS</td>\n",
       "      <td>2015-07-26T00:03:30.000Z</td>\n",
       "      <td>2016-06-02T01:58:12.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>crossref</td>\n",
       "      <td>CrossRef</td>\n",
       "      <td>2014-08-03T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>scitech</td>\n",
       "      <td>DoE's SciTech Connect Database</td>\n",
       "      <td>2014-10-03T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>figshare</td>\n",
       "      <td>figshare</td>\n",
       "      <td>2014-10-28T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T15:07:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pubmedcentral</td>\n",
       "      <td>PubMed Central</td>\n",
       "      <td>2014-12-28T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>dataone</td>\n",
       "      <td>DataONE: Data Observation Network for Earth</td>\n",
       "      <td>2015-04-11T00:00:00.000Z</td>\n",
       "      <td>2016-05-28T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>arxiv_oai</td>\n",
       "      <td>ArXiv</td>\n",
       "      <td>2014-10-03T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rcaap</td>\n",
       "      <td>RCAAP - Repositório Científico de Acesso Abert...</td>\n",
       "      <td>2015-12-27T02:00:54.000Z</td>\n",
       "      <td>2016-06-03T05:08:35.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>cyberleninka</td>\n",
       "      <td>CyberLeninka - Russian open access scientific ...</td>\n",
       "      <td>2015-12-22T00:00:00.000Z</td>\n",
       "      <td>2016-06-02T22:07:47.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>citeseerx</td>\n",
       "      <td>CiteSeerX Scientific Literature Digital Librar...</td>\n",
       "      <td>2008-07-01T00:00:00.000Z</td>\n",
       "      <td>2016-06-02T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>bhl</td>\n",
       "      <td>Biodiversity Heritage Library OAI Repository</td>\n",
       "      <td>2015-04-19T19:06:17.000Z</td>\n",
       "      <td>2016-06-02T12:18:37.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>springer</td>\n",
       "      <td>Springer</td>\n",
       "      <td>2015-11-11T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>clinicaltrials</td>\n",
       "      <td>ClinicalTrials.gov</td>\n",
       "      <td>2014-10-07T00:00:00.000Z</td>\n",
       "      <td>2016-06-02T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>uiucideals</td>\n",
       "      <td>University of Illinois at Urbana-Champaign, ID...</td>\n",
       "      <td>2015-01-22T19:53:34.000Z</td>\n",
       "      <td>2016-06-02T21:00:13.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>mit</td>\n",
       "      <td>DSpace@MIT</td>\n",
       "      <td>2014-10-04T04:25:20.000Z</td>\n",
       "      <td>2016-06-03T19:02:54.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>plos</td>\n",
       "      <td>Public Library of Science</td>\n",
       "      <td>2015-04-11T00:00:00.000Z</td>\n",
       "      <td>2016-06-02T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>iowaresearch</td>\n",
       "      <td>Iowa Research Online</td>\n",
       "      <td>2011-07-09T12:38:13.000Z</td>\n",
       "      <td>2016-06-01T21:56:10.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>lshtm</td>\n",
       "      <td>London School of Hygiene and Tropical Medicine...</td>\n",
       "      <td>2015-11-17T16:21:22.000Z</td>\n",
       "      <td>2016-06-02T14:28:35.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>krex</td>\n",
       "      <td>K-State Research Exchange</td>\n",
       "      <td>2016-01-25T15:24:52.000Z</td>\n",
       "      <td>2016-06-02T13:54:24.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ucescholarship</td>\n",
       "      <td>eScholarship @ University of California</td>\n",
       "      <td>2013-11-26T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>biomedcentral</td>\n",
       "      <td>BioMed Central</td>\n",
       "      <td>2015-02-04T00:00:00.000Z</td>\n",
       "      <td>2016-06-03T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>zenodo</td>\n",
       "      <td>Zenodo</td>\n",
       "      <td>2015-04-21T13:07:43.000Z</td>\n",
       "      <td>2016-06-02T21:43:18.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>shareok</td>\n",
       "      <td>SHAREOK Repository</td>\n",
       "      <td>2015-01-08T02:01:20.000Z</td>\n",
       "      <td>2016-06-02T07:08:46.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>cambridge</td>\n",
       "      <td>Apollo @ University of Cambridge</td>\n",
       "      <td>2015-10-26T10:00:15.000Z</td>\n",
       "      <td>2016-06-03T15:00:57.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>noaa_nodc</td>\n",
       "      <td>National Oceanographic Data Center</td>\n",
       "      <td>2015-09-30T20:18:16.000Z</td>\n",
       "      <td>2016-06-03T19:11:04.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>utaustin</td>\n",
       "      <td>University of Texas at Austin Digital Repository</td>\n",
       "      <td>2014-10-24T16:45:59.000Z</td>\n",
       "      <td>2016-06-02T22:47:09.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>caltech</td>\n",
       "      <td>CaltechAUTHORS</td>\n",
       "      <td>2015-04-19T14:25:00.000Z</td>\n",
       "      <td>2016-06-03T22:47:04.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>uwashington</td>\n",
       "      <td>ResearchWorks @ University of Washington</td>\n",
       "      <td>2015-02-04T11:01:48.000Z</td>\n",
       "      <td>2016-06-02T22:35:30.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ukansas</td>\n",
       "      <td>KU ScholarWorks</td>\n",
       "      <td>2016-02-17T09:01:53.000Z</td>\n",
       "      <td>2016-06-03T08:01:32.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>vtech</td>\n",
       "      <td>Virginia Tech VTechWorks</td>\n",
       "      <td>2014-11-29T07:00:11.000Z</td>\n",
       "      <td>2016-06-01T16:07:01.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>trinity</td>\n",
       "      <td>Digital Commons @ Trinity University</td>\n",
       "      <td>2014-11-06T20:52:00.000Z</td>\n",
       "      <td>2016-05-31T16:56:04.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>uhawaii</td>\n",
       "      <td>ScholarSpace at University of Hawaii at Manoa</td>\n",
       "      <td>2016-04-01T04:05:45.000Z</td>\n",
       "      <td>2016-06-02T21:39:18.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>texasstate</td>\n",
       "      <td>DSpace at Texas State University</td>\n",
       "      <td>2014-12-06T00:00:12.000Z</td>\n",
       "      <td>2016-05-20T20:52:16.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>udc</td>\n",
       "      <td>University of Minnesota, Digital Conservancy</td>\n",
       "      <td>2016-04-14T14:31:37.000Z</td>\n",
       "      <td>2016-06-02T15:00:35.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>valposcholar</td>\n",
       "      <td>Valparaiso University ValpoScholar</td>\n",
       "      <td>2015-01-22T21:35:08.000Z</td>\n",
       "      <td>2016-06-02T22:45:58.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>umassmed</td>\n",
       "      <td>eScholarship@UMMS</td>\n",
       "      <td>2016-03-15T19:41:00.000Z</td>\n",
       "      <td>2016-05-31T17:27:57.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>neurovault</td>\n",
       "      <td>NeuroVault.org</td>\n",
       "      <td>2013-07-03T10:48:50.369Z</td>\n",
       "      <td>2015-11-04T16:41:41.220Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>spdataverse</td>\n",
       "      <td>Scholars Portal dataverse</td>\n",
       "      <td>2014-11-13T18:00:01.000Z</td>\n",
       "      <td>2016-06-02T18:00:01.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>pcom</td>\n",
       "      <td>DigitalCommons@PCOM</td>\n",
       "      <td>2015-08-31T13:28:12.000Z</td>\n",
       "      <td>2016-05-31T16:50:07.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>mla</td>\n",
       "      <td>MLA Commons</td>\n",
       "      <td>2015-05-04T16:57:02.000Z</td>\n",
       "      <td>2016-06-02T23:36:20.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>ut_chattanooga</td>\n",
       "      <td>University of Tennessee at Chattanooga</td>\n",
       "      <td>2015-01-23T15:27:16.000Z</td>\n",
       "      <td>2016-06-02T18:40:12.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>uwo</td>\n",
       "      <td>Western University</td>\n",
       "      <td>2016-04-19T16:07:36.000Z</td>\n",
       "      <td>2016-06-03T19:59:45.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>triceratops</td>\n",
       "      <td>Triceratops: Tri-College Digital Repository</td>\n",
       "      <td>2016-05-17T02:00:53.000Z</td>\n",
       "      <td>2016-06-02T02:01:53.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>iwu_commons</td>\n",
       "      <td>Digital Commons @ Illinois Wesleyan University</td>\n",
       "      <td>2016-01-26T16:47:20.000Z</td>\n",
       "      <td>2016-06-03T17:11:57.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>utktrace</td>\n",
       "      <td>Trace: Tennessee Research and Creative Exchange</td>\n",
       "      <td>2015-10-26T21:01:08.000Z</td>\n",
       "      <td>2016-06-01T17:50:28.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>boise_state</td>\n",
       "      <td>Boise State University ScholarWorks</td>\n",
       "      <td>2016-05-16T16:12:08.000Z</td>\n",
       "      <td>2016-06-03T19:39:50.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>mason</td>\n",
       "      <td>Mason Archival Repository Service</td>\n",
       "      <td>2015-11-16T20:22:33.000Z</td>\n",
       "      <td>2016-05-24T23:53:10.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>fit</td>\n",
       "      <td>Florida Institute of Technology</td>\n",
       "      <td>2016-02-22T16:59:19.000Z</td>\n",
       "      <td>2016-05-24T18:47:03.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>stcloud</td>\n",
       "      <td>The repository at St Cloud State</td>\n",
       "      <td>2014-12-03T22:38:10.000Z</td>\n",
       "      <td>2016-04-22T01:01:34.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>csir</td>\n",
       "      <td>CSIR Researchspace</td>\n",
       "      <td>2016-04-20T21:55:15.000Z</td>\n",
       "      <td>2016-05-30T21:55:15.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>iu</td>\n",
       "      <td>Indiana University Libraries' IUScholarWorks</td>\n",
       "      <td>2016-04-19T14:48:25.000Z</td>\n",
       "      <td>2016-06-02T19:34:20.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>cogprints</td>\n",
       "      <td>Cognitive Sciences ePrint Archive</td>\n",
       "      <td>2015-04-20T11:26:30.000Z</td>\n",
       "      <td>2016-03-24T19:13:57.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>usgs</td>\n",
       "      <td>United States Geological Survey</td>\n",
       "      <td>2016-02-23T09:24:26.000Z</td>\n",
       "      <td>2016-02-25T10:45:56.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>hacettepe</td>\n",
       "      <td>Hacettepe University DSpace on LibLiveCD</td>\n",
       "      <td>2014-05-23T23:00:08.000Z</td>\n",
       "      <td>2014-08-26T23:00:09.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>lwbin</td>\n",
       "      <td>Lake Winnipeg Basin Information Network</td>\n",
       "      <td>2014-11-12T18:04:57.746Z</td>\n",
       "      <td>2015-10-26T13:10:08.862Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>purdue</td>\n",
       "      <td>PURR - Purdue University Research Repository</td>\n",
       "      <td>2015-12-30T05:00:00.000Z</td>\n",
       "      <td>2016-05-17T05:59:24.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>sldr</td>\n",
       "      <td>Speech and Language Data Repository (SLDR/ORTO...</td>\n",
       "      <td>2015-12-15T00:00:00.000Z</td>\n",
       "      <td>2016-04-20T00:00:00.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>ncar</td>\n",
       "      <td>Earth System Grid at NCAR</td>\n",
       "      <td>2015-10-30T18:28:29.000Z</td>\n",
       "      <td>2016-04-05T17:53:18.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>ucsd</td>\n",
       "      <td>UC San Diego</td>\n",
       "      <td>1961-01-01T00:00:00.000Z</td>\n",
       "      <td>2016-05-24T22:20:06.000Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>nist</td>\n",
       "      <td>NIST MaterialsData</td>\n",
       "      <td>2015-11-02T22:04:55.000Z</td>\n",
       "      <td>2015-11-19T12:30:06.000Z</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>116 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    source_shortname                                    source_longname  \\\n",
       "0           datacite                                       DataCite MDS   \n",
       "1           crossref                                           CrossRef   \n",
       "2            scitech                     DoE's SciTech Connect Database   \n",
       "3           figshare                                           figshare   \n",
       "4      pubmedcentral                                     PubMed Central   \n",
       "5            dataone        DataONE: Data Observation Network for Earth   \n",
       "6          arxiv_oai                                              ArXiv   \n",
       "7              rcaap  RCAAP - Repositório Científico de Acesso Abert...   \n",
       "8       cyberleninka  CyberLeninka - Russian open access scientific ...   \n",
       "9          citeseerx  CiteSeerX Scientific Literature Digital Librar...   \n",
       "10               bhl       Biodiversity Heritage Library OAI Repository   \n",
       "11          springer                                           Springer   \n",
       "12    clinicaltrials                                 ClinicalTrials.gov   \n",
       "13        uiucideals  University of Illinois at Urbana-Champaign, ID...   \n",
       "14               mit                                         DSpace@MIT   \n",
       "15              plos                          Public Library of Science   \n",
       "16      iowaresearch                               Iowa Research Online   \n",
       "17             lshtm  London School of Hygiene and Tropical Medicine...   \n",
       "18              krex                          K-State Research Exchange   \n",
       "19    ucescholarship            eScholarship @ University of California   \n",
       "20     biomedcentral                                     BioMed Central   \n",
       "21            zenodo                                             Zenodo   \n",
       "22           shareok                                 SHAREOK Repository   \n",
       "23         cambridge                   Apollo @ University of Cambridge   \n",
       "24         noaa_nodc                 National Oceanographic Data Center   \n",
       "25          utaustin   University of Texas at Austin Digital Repository   \n",
       "26           caltech                                     CaltechAUTHORS   \n",
       "27       uwashington           ResearchWorks @ University of Washington   \n",
       "28           ukansas                                    KU ScholarWorks   \n",
       "29             vtech                           Virginia Tech VTechWorks   \n",
       "..               ...                                                ...   \n",
       "86           trinity               Digital Commons @ Trinity University   \n",
       "87           uhawaii      ScholarSpace at University of Hawaii at Manoa   \n",
       "88        texasstate                   DSpace at Texas State University   \n",
       "89               udc       University of Minnesota, Digital Conservancy   \n",
       "90      valposcholar                 Valparaiso University ValpoScholar   \n",
       "91          umassmed                                  eScholarship@UMMS   \n",
       "92        neurovault                                     NeuroVault.org   \n",
       "93       spdataverse                          Scholars Portal dataverse   \n",
       "94              pcom                                DigitalCommons@PCOM   \n",
       "95               mla                                        MLA Commons   \n",
       "96    ut_chattanooga             University of Tennessee at Chattanooga   \n",
       "97               uwo                                 Western University   \n",
       "98       triceratops        Triceratops: Tri-College Digital Repository   \n",
       "99       iwu_commons     Digital Commons @ Illinois Wesleyan University   \n",
       "100         utktrace    Trace: Tennessee Research and Creative Exchange   \n",
       "101      boise_state                Boise State University ScholarWorks   \n",
       "102            mason                  Mason Archival Repository Service   \n",
       "103              fit                    Florida Institute of Technology   \n",
       "104          stcloud                   The repository at St Cloud State   \n",
       "105             csir                                 CSIR Researchspace   \n",
       "106               iu       Indiana University Libraries' IUScholarWorks   \n",
       "107        cogprints                  Cognitive Sciences ePrint Archive   \n",
       "108             usgs                    United States Geological Survey   \n",
       "109        hacettepe           Hacettepe University DSpace on LibLiveCD   \n",
       "110            lwbin            Lake Winnipeg Basin Information Network   \n",
       "111           purdue       PURR - Purdue University Research Repository   \n",
       "112             sldr  Speech and Language Data Repository (SLDR/ORTO...   \n",
       "113             ncar                          Earth System Grid at NCAR   \n",
       "114             ucsd                                       UC San Diego   \n",
       "115             nist                                 NIST MaterialsData   \n",
       "\n",
       "                earliest_date               latest_date  \n",
       "0    2015-07-26T00:03:30.000Z  2016-06-02T01:58:12.000Z  \n",
       "1    2014-08-03T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "2    2014-10-03T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "3    2014-10-28T00:00:00.000Z  2016-06-03T15:07:00.000Z  \n",
       "4    2014-12-28T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "5    2015-04-11T00:00:00.000Z  2016-05-28T00:00:00.000Z  \n",
       "6    2014-10-03T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "7    2015-12-27T02:00:54.000Z  2016-06-03T05:08:35.000Z  \n",
       "8    2015-12-22T00:00:00.000Z  2016-06-02T22:07:47.000Z  \n",
       "9    2008-07-01T00:00:00.000Z  2016-06-02T00:00:00.000Z  \n",
       "10   2015-04-19T19:06:17.000Z  2016-06-02T12:18:37.000Z  \n",
       "11   2015-11-11T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "12   2014-10-07T00:00:00.000Z  2016-06-02T00:00:00.000Z  \n",
       "13   2015-01-22T19:53:34.000Z  2016-06-02T21:00:13.000Z  \n",
       "14   2014-10-04T04:25:20.000Z  2016-06-03T19:02:54.000Z  \n",
       "15   2015-04-11T00:00:00.000Z  2016-06-02T00:00:00.000Z  \n",
       "16   2011-07-09T12:38:13.000Z  2016-06-01T21:56:10.000Z  \n",
       "17   2015-11-17T16:21:22.000Z  2016-06-02T14:28:35.000Z  \n",
       "18   2016-01-25T15:24:52.000Z  2016-06-02T13:54:24.000Z  \n",
       "19   2013-11-26T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "20   2015-02-04T00:00:00.000Z  2016-06-03T00:00:00.000Z  \n",
       "21   2015-04-21T13:07:43.000Z  2016-06-02T21:43:18.000Z  \n",
       "22   2015-01-08T02:01:20.000Z  2016-06-02T07:08:46.000Z  \n",
       "23   2015-10-26T10:00:15.000Z  2016-06-03T15:00:57.000Z  \n",
       "24   2015-09-30T20:18:16.000Z  2016-06-03T19:11:04.000Z  \n",
       "25   2014-10-24T16:45:59.000Z  2016-06-02T22:47:09.000Z  \n",
       "26   2015-04-19T14:25:00.000Z  2016-06-03T22:47:04.000Z  \n",
       "27   2015-02-04T11:01:48.000Z  2016-06-02T22:35:30.000Z  \n",
       "28   2016-02-17T09:01:53.000Z  2016-06-03T08:01:32.000Z  \n",
       "29   2014-11-29T07:00:11.000Z  2016-06-01T16:07:01.000Z  \n",
       "..                        ...                       ...  \n",
       "86   2014-11-06T20:52:00.000Z  2016-05-31T16:56:04.000Z  \n",
       "87   2016-04-01T04:05:45.000Z  2016-06-02T21:39:18.000Z  \n",
       "88   2014-12-06T00:00:12.000Z  2016-05-20T20:52:16.000Z  \n",
       "89   2016-04-14T14:31:37.000Z  2016-06-02T15:00:35.000Z  \n",
       "90   2015-01-22T21:35:08.000Z  2016-06-02T22:45:58.000Z  \n",
       "91   2016-03-15T19:41:00.000Z  2016-05-31T17:27:57.000Z  \n",
       "92   2013-07-03T10:48:50.369Z  2015-11-04T16:41:41.220Z  \n",
       "93   2014-11-13T18:00:01.000Z  2016-06-02T18:00:01.000Z  \n",
       "94   2015-08-31T13:28:12.000Z  2016-05-31T16:50:07.000Z  \n",
       "95   2015-05-04T16:57:02.000Z  2016-06-02T23:36:20.000Z  \n",
       "96   2015-01-23T15:27:16.000Z  2016-06-02T18:40:12.000Z  \n",
       "97   2016-04-19T16:07:36.000Z  2016-06-03T19:59:45.000Z  \n",
       "98   2016-05-17T02:00:53.000Z  2016-06-02T02:01:53.000Z  \n",
       "99   2016-01-26T16:47:20.000Z  2016-06-03T17:11:57.000Z  \n",
       "100  2015-10-26T21:01:08.000Z  2016-06-01T17:50:28.000Z  \n",
       "101  2016-05-16T16:12:08.000Z  2016-06-03T19:39:50.000Z  \n",
       "102  2015-11-16T20:22:33.000Z  2016-05-24T23:53:10.000Z  \n",
       "103  2016-02-22T16:59:19.000Z  2016-05-24T18:47:03.000Z  \n",
       "104  2014-12-03T22:38:10.000Z  2016-04-22T01:01:34.000Z  \n",
       "105  2016-04-20T21:55:15.000Z  2016-05-30T21:55:15.000Z  \n",
       "106  2016-04-19T14:48:25.000Z  2016-06-02T19:34:20.000Z  \n",
       "107  2015-04-20T11:26:30.000Z  2016-03-24T19:13:57.000Z  \n",
       "108  2016-02-23T09:24:26.000Z  2016-02-25T10:45:56.000Z  \n",
       "109  2014-05-23T23:00:08.000Z  2014-08-26T23:00:09.000Z  \n",
       "110  2014-11-12T18:04:57.746Z  2015-10-26T13:10:08.862Z  \n",
       "111  2015-12-30T05:00:00.000Z  2016-05-17T05:59:24.000Z  \n",
       "112  2015-12-15T00:00:00.000Z  2016-04-20T00:00:00.000Z  \n",
       "113  2015-10-30T18:28:29.000Z  2016-04-05T17:53:18.000Z  \n",
       "114  1961-01-01T00:00:00.000Z  2016-05-24T22:20:06.000Z  \n",
       "115  2015-11-02T22:04:55.000Z  2015-11-19T12:30:06.000Z  \n",
       "\n",
       "[116 rows x 4 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# What's the earliest and latest document from each source?\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "date_stats_agg = {\n",
    "    \"aggregations\": {\n",
    "        \"sources\": {\n",
    "            \"terms\": {\"field\": \"_type\", \"size\": 0},\n",
    "            \"aggregations\": {\n",
    "                \"source_stats\": {\n",
    "                    \"stats\": {\"field\": \"providerUpdatedDateTime\"}\n",
    "                }\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "date_results = query_share(SHARE_SEARCH_API, date_stats_agg)['aggregations']['sources']['buckets']\n",
    "\n",
    "date_results_df = pd.DataFrame()\n",
    "date_results_df['source_shortname'] = [result['key'].encode('utf-8') for result in date_results]\n",
    "date_results_df['source_longname'] = [get_longname_for_shortname(name).encode('utf-8') for name in date_results_df['source_shortname']]\n",
    "date_results_df['earliest_date'] = [result['source_stats']['min_as_string'] for result in date_results]\n",
    "date_results_df['latest_date'] = [result['source_stats']['max_as_string'] for result in date_results]\n",
    "date_results_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Uncomment the following lines if running locally - will save to file formats\n",
    "\n",
    "# date_results_df.to_csv('SHARE_Min_Max_dates.csv')\n",
    "# date_results_df.to_excel('SHARE_Min_Max_dates.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Lucene Search and NOT Queries\n",
    "\n",
    "A user wanted to know how to query for one term but exclude another"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "query = '?q=pedigree NOT child'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "results = requests.get(SHARE_SEARCH_API + query).json()\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Querying by Document Type\n",
    "\n",
    "Currently, document type is not curated by SHARE. However, we do collected many sources that are using the OAI-PMH metadata protocol, which includes dc:type. You can search that field in SHARE for now, until the harvesters collect and curate document type."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "query = '?q=otherProperties.properties.type:article'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "results = requests.get(SHARE_SEARCH_API + query).json()\n",
    "\n",
    "for result in results['results']:\n",
    "    for prop in result['otherProperties']:\n",
    "        if prop['name'] == 'type':\n",
    "            print(prop)\n",
    "    print(result['title'])\n",
    "    print(result['uris']['canonicalUri'])\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Here is an analysis of the top terms found in SHARE's collected dc:type field"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from sharepa import ShareSearch, basic_search\n",
    "from sharepa.helpers import pretty_print\n",
    "\n",
    "type_search = ShareSearch()\n",
    "total_documents = basic_search.count()\n",
    "\n",
    "type_search.aggs.bucket(\n",
    "    'typeTermFilter',  # Every aggregation needs a name\n",
    "    'terms',  # There are many kinds of aggregations\n",
    "    field='otherProperties.properties.type',\n",
    "    exclude= \"of|and|or\",\n",
    "    size=50,\n",
    ")\n",
    "\n",
    "type_results_executed = type_search.execute()\n",
    "\n",
    "type_results = type_results_executed.aggregations.typeTermFilter.to_dict()['buckets']\n",
    "\n",
    "type_dataframe = pd.DataFrame(type_results)\n",
    "type_dataframe['percent'] = (type_dataframe['doc_count'] / total_documents)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "type_dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Query by Exact Phrase\n",
    "\n",
    "Question -- Is there a way to search SHARE for a specific phrase? For example, information literacy, information AND literacy, and \"information literacy\" give results with both terms, but not necessarily as the phrase \"information literacy.\" Information and literacy can be in different parts of the record."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "phrase_query = {\n",
    "    \"query\": {\n",
    "        \"match_phrase\" : {\n",
    "            \"title\" : \"information literacy\"\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "results = query_share(SHARE_SEARCH_API, phrase_query)\n",
    "\n",
    "for result in results['results']:\n",
    "    print(\n",
    "        '{} -- from {} -- {}'.format(\n",
    "            result['title'].encode('utf-8'),\n",
    "            result['shareProperties']['source'].encode('utf-8'),\n",
    "            result['uris']['canonicalUri'].encode('utf-8')\n",
    "        )\n",
    "    )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "# Using sharepa\n",
    "\n",
    "phrase_search = ShareSearch()\n",
    "\n",
    "phrase_search = phrase_search.query(\n",
    "    'match_phrase',\n",
    "    title=\"information literacy\"\n",
    ")\n",
    "\n",
    "results = phrase_search.execute()\n",
    "\n",
    "for result in results:\n",
    "    print(\n",
    "        '{} -- from {} -- {}'.format(\n",
    "            result.title.encode('utf-8'),\n",
    "            result.shareProperties.source.encode('utf-8'),\n",
    "            result.uris.canonicalUri.encode('utf-8')\n",
    "        )\n",
    "    )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Checking the Validity of our Results\n",
    "\n",
    "If we make a query, we want to do a quick run through of our results just to make sure the query is behaving as we think it is.\n",
    "\n",
    "Here's an example of making a long query to include and not include specific results. We'll then iterate through all of the results and make sure that the they contain the things we want and do not contain the things we don't want."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "query_url = 'https://osf.io/api/v1/share/search/?start=0&size=1000&q=pedigree* OR \"relatedness matrix\" NOT child NOT \"family tree\" NOT \"dog food\" NOT diagnosis'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "desired_words = ['pedigree', 'relatedness matrix']\n",
    "undesired_words = ['child', 'family tree', 'dog food', 'diagnosis']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# requests.get(query_url).json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Iterate through those results and make sure the titles and descriptions have at least one desired word, \n",
    "# and do not have any of the undesired words\n",
    "\n",
    "\n",
    "def results_have_correct_words(this_page, desired, undesired):\n",
    "    print('REJECTING {}'.format(undesired))\n",
    "    title_check = True\n",
    "    description_check = True\n",
    "    number_checked = 0\n",
    "    for result in this_page:\n",
    "        number_checked += 1\n",
    "        if any(word in result['title'] for word in undesired) or not any(word in result['title'] for word in desired):\n",
    "            title_check = False\n",
    "        if result.get('description'):\n",
    "            if any(word in result['description'] for word in undesired) or not any(word in result['description'] for word in desired):\n",
    "                description_check = False\n",
    "        if not title_check and not description_check:\n",
    "            print('-------------')\n",
    "            print(result['title'])\n",
    "            print(result['description'])\n",
    "            print('title: {}  description: {}'.format(title_check, description_check))\n",
    "    return title_check or description_check, number_checked\n",
    "\n",
    "\n",
    "total_results = requests.get(query_url).json()['count']\n",
    "total_checked = 0\n",
    "start = 0\n",
    "size = 1000\n",
    "def check_results(url_to_check, total_checked, start, desired, undesired):\n",
    "    print('starting at {}'.format(start))\n",
    "    url = furl.furl(url_to_check)\n",
    "    url.args['start'] = start\n",
    "    results = requests.get(url.url).json()\n",
    "    if total_checked < total_results:\n",
    "        status, checked_this_time = results_have_correct_words(results['results'], desired_words, undesired_words)\n",
    "        total_checked += checked_this_time\n",
    "        print('I checked {} this time'.format(checked_this_time))\n",
    "\n",
    "        if status == False:\n",
    "            return False\n",
    "        else:\n",
    "            start += size\n",
    "            check_results(url, total_checked, start, desired, undesired)\n",
    "    \n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "starting at 0\n",
      "REJECTING ['child', 'family tree', 'dog food', 'diagnosis']\n",
      "-------------\n",
      "Hybrid_Matrix\n",
      "Relatedness matrix used by the \"Hybrid\" method at the AsREML format (.grm). Matrix generated a posteriori with a R script.\n",
      "title: False  description: False\n",
      "-------------\n",
      "Relatedness_Matrix\n",
      "Relatedness matrix used by the \"Relatedness\" method at the AsREML format (.grm). Matrix generated a posteriori with a R script.\n",
      "title: False  description: False\n",
      "-------------\n",
      "HSFS_Matrix\n",
      "Relatedness matrix used by the \"HSFS\" method at the AsREML format (.grm). Matrix generated a posteriori with a R script.\n",
      "title: False  description: False\n",
      "-------------\n",
      "HSFS_Matrix\n",
      "Relatedness matrix used by the \"HSFS\" method at the AsREML format (.grm). Matrix generated a posteriori with a R script.\n",
      "title: False  description: False\n",
      "-------------\n",
      "PhiA_Matrix\n",
      "Relatedness matrix used by the reference method \"PhiA\" at the AsREML format (.grm). Matrix generated a posteriori with a R script.\n",
      "title: False  description: False\n",
      "-------------\n",
      "PhiA_Matrix\n",
      "Relatedness matrix used by the reference method \"PhiA\" at the AsREML format (.grm). Matrix generated a posteriori with a R script.\n",
      "title: False  description: False\n",
      "-------------\n",
      "Heritability Estimation in Matrix-Variate Mixed models -- A Bayesian\n",
      "  Approach\n",
      "  Since the emergence of genome-wide association studies (GWASs), estimation of\n",
      "the narrow sense heritability explained by common single-nucleotide\n",
      "polymorphisms (SNPs) via linear mixed model approaches became widely used. As\n",
      "in most GWASs, most of the heritability analyses are performed using univariate\n",
      "approaches i.e. considering each phenotype independently. In this study, we\n",
      "propose a Bayesian matrix-variate mixed model that takes into account the\n",
      "genetic correlation between phenotypes in addition to the genetic correlation\n",
      "between individuals which is usually modelled via a relatedness matrix. We\n",
      "showed that when the relatedness matrix is estimated using all the genome-wide\n",
      "SNPs, our model is equivalent to a matrix normal regression with matrix normal\n",
      "prior on the effect sizes. Using real data we demonstrate that there is a boost\n",
      "in the heritability explained when phenotypes are jointly modelled (25-35%\n",
      "increase). In fact based on their standard error, the joint modelling provides\n",
      "more accurate estimates of the heritability over the univariate modelling.\n",
      "Moreover, our Bayesian approach provides slightly higher estimates of\n",
      "heritability compared to the maximum likelihood method. On the other hand,\n",
      "although our method performs less well in phenotype prediction, we note that an\n",
      "initial imputation step relatively increases the prediction accuracy.\n",
      "\n",
      "title: False  description: False\n",
      "-------------\n",
      "Limitations of GCTA as a solution to the missing heritability problem\n",
      "The genetic contribution to a phenotype is frequently measured by heritability, the fraction of trait variation explained by genetic differences. Hundreds of publications have found DNA polymorphisms that are statistically associated with diseases or quantitative traits [genome-wide association studies (GWASs)]. Genome-wide complex trait analysis (GCTA), a recent method of analyzing such data, finds high heritabilities for such phenotypes. We analyze GCTA and show that the heritability estimates it produces are highly sensitive to the structure of the genetic relatedness matrix, to the sampling of phenotypes and subjects, and to the accuracy of phenotype measurements. Plausible modifications of the method aimed at increasing stability yield much smaller heritabilities. It is essential to reevaluate the many published heritability estimates based on GCTA.\n",
      "title: False  description: False\n",
      "-------------\n",
      "A Kinship-Based Modification of the Armitage Trend Test to Address Hidden Population Structure and Small Differential Genotyping Errors\n",
      "Background/Aims We propose a modification of the well-known Armitage trend test to address the problems associated with hidden population structure and hidden relatedness in genome-wide case-control association studies.\n",
      "Methods The new test adopts beneficial traits from three existing testing strategies: the principal components, mixed model, and genomic control while avoiding some of their disadvantageous characteristics, such as the tendency of the principal components method to over-correct in certain situations or the failure of the genomic control approach to reorder the adjusted tests based on their degree of alignment with the underlying hidden structure. The new procedure is based on Gauss-Markov estimators derived from a straightforward linear model with an imposed variance structure proportional to an empirical relatedness matrix. Lastly, conceptual and analytical similarities to and distinctions from other approaches are emphasized throughout.\n",
      "Results Our simulations show that the power performance of the proposed test is quite promising compared to the considered competing strategies. The power gains are especially large when small differential differences between cases and controls are present; a likely scenario when public controls are used in multiple studies.\n",
      "Conclusion The proposed modified approach attains high power more consistently than that of the existing commonly implemented tests. Its performance improvement is most apparent when small but detectable systematic differences between cases and controls exist.\n",
      "title: False  description: False\n",
      "-------------\n",
      "Dominant Genetic Variation and Missing Heritability for Human Complex Traits: Insights from Twin versus Genome-wide Common SNP Models\n",
      "In order to further illuminate the potential role of dominant genetic variation in the âmissing heritabilityâ debate, we investigated the additive (narrow-sense heritability, h2) and dominant (Î´2) genetic variance for 18 human complex traits. Within the same study base (10,682 Swedish twins), we calculated and compared the estimates from classic twin-based structural equation model with SNP-based genomic-relatedness-matrix restricted maximum likelihood [GREML(d)] method. Contributions of Î´2 were evident for 14 traits in twin models (average Î´2twin = 0.25, range 0.14â0.49), two of which also displayed significant Î´2 in the GREMLd analyses (triglycerides Î´2SNP = 0.28 and waist circumference Î´2SNP = 0.19). On average, the proportion of h2SNP/h2twin was 70% for ADE-fitted traits (for which the best-fitting model included additive and dominant genetic and unique environmental components) and 31% for AE-fitted traits (for which the best-fitting model included additive genetic and unique environmental components). Independent evidence for contribution from shared environment, also in ADE-fitted traits, was obtained from self-reported within-pair contact frequency and age at separation. We conclude that despite the fact that additive genetics appear to constitute the bulk of genetic influences for most complex traits, dominant genetic variation might often be masked by shared environment in twin and family studies and might therefore have a more prominent role than what family-based estimates often suggest. The risk of erroneously attributing all inherited genetic influences (additive and dominant) to the h2 in too-small twin studies might also lead to exaggerated âmissing heritabilityâ (the proportion of h2 that remains unexplained by SNPs).\n",
      "title: False  description: False\n",
      "-------------\n",
      "Locating a genetic marker for rheumatic fever using a pedigree analysis\n",
      "Includes bibliographical references (pages [58]-61)\n",
      "title: False  description: False\n",
      "-------------\n",
      "Global pedigree disequilibrium test at 1q32–42 in 332 families with systemic lupus erythematosus\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'description'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-78-da42e0c6c761>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcheck_results\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdesired_words\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mundesired_words\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-77-d71ba27b5c1e>\u001b[0m in \u001b[0;36mcheck_results\u001b[0;34m(url_to_check, total_checked, start, desired, undesired)\u001b[0m\n\u001b[1;32m     33\u001b[0m     \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     34\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mtotal_checked\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mtotal_results\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 35\u001b[0;31m         \u001b[0mstatus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchecked_this_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresults_have_correct_words\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'results'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdesired_words\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mundesired_words\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     36\u001b[0m         \u001b[0mtotal_checked\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mchecked_this_time\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     37\u001b[0m         \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'I checked {} this time'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchecked_this_time\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-77-d71ba27b5c1e>\u001b[0m in \u001b[0;36mresults_have_correct_words\u001b[0;34m(this_page, desired, undesired)\u001b[0m\n\u001b[1;32m     18\u001b[0m             \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'-------------'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m             \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'title'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 20\u001b[0;31m             \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'description'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     21\u001b[0m             \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'title: {}  description: {}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtitle_check\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdescription_check\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mtitle_check\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mdescription_check\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnumber_checked\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'description'"
     ]
    }
   ],
   "source": [
    "res = check_results(query_url, 0, 0, desired_words, undesired_words)\n",
    "res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
